### PR TITLE
VIH-9353 - Team leader manages team search for user

### PR DESCRIPTION
**Pull request by marcogagliardi**
https://tools.hmcts.net/jira/browse/VIH-9353

added search functionality

image

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts [hunk 1]
```diff
@@ -1,7 +1,81 @@
...
+    message: string;
+    users: JusticeUserResponse[];
+    form: FormGroup;
+    isEditing = false;
+    isSaving = false;
+    error = false;
+    displayAddButton = false;
+    errorMessage = false;
+
+    editUser(id) {
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:35]: As this method is currently empty, could we remove it for now?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts [hunk 2]
```diff
@@ -1,7 +1,81 @@
...
+        // TODO: edit will enable all the inputs in a row.
+        // const row = this.elRef.nativeElement.getElementsByClassName(id) as HTMLCollection;
+        // for (let i = 0; i < row.length; i++) {
+        //     (<HTMLElement>row[i]).removeAttribute('disabled');
+        // }
+    }
+
+    deleteUser(id) {}
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:43]: as per previous comment, think we can remove this for now as it is empty

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts [hunk 3]
```diff
@@ -1,7 +1,81 @@
...
+                }
+                this.displayAddButton = true;
+            },
+            error => {
+                this.handleListError(error, 'users');
+            }
+        );
+    }
+
+    addUsers() {}
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:75]: as per previous comment, think we can remove this for now as it is empty

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts [hunk 4]
```diff
@@ -1,7 +1,81 @@
...
+        this.errorMessage = false;
+        this.displayAddButton = false;
+        this.displayMessage = false;
+        this.message = '';
+        this.isEditing = false;
+        this.videoHearingService.getUsers(term).subscribe(
+            (data: JusticeUserResponse[]) => {
+                this.users = data;
+                this.logger.debug(`${this.loggerPrefix} Updating list of users.`, { users: data.length });
+                if (this.users.length > 20) {
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:56]: we could replace 20 with the variable here

`if (this.users.length > this.filterSize)`

File: AdminWebsite/AdminWebsite/Controllers/UserIdentityController.cs [hunk 5]
```diff
@@ -54,20 +54,21 @@ public async Task<ActionResult<UserProfileResponse>> GetUserProfile()
         }
         
         /// <summary>
-        /// Get the Justice User list from the JusticeUser table
+        /// Get list of Justice User filtered by passed term. If term is null then not filter applied.
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/Controllers/UserIdentityController.cs:57]: could we tweak this description slightly:

`Get list of Justice User filtered by term. If term is null then no filter applied.`

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.html [hunk 6]
```diff
@@ -2,15 +2,108 @@
...
+            class="{{ user.id }}"
+            required="true"
+          />
+        </td>
+        <td class="govuk-table__cell">
+          <button
+            class="govuk-button govuk-!-margin-right-6"
+            data-module="govuk-button"
+            id="editButton_{{ user.id }}"
+            (click)="editUser(user.id)"
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.html:96]: think we can remove this click handler until we implement edit + delete

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.html [hunk 7]
```diff
@@ -2,15 +2,108 @@
...
+          >
+            Edit
+          </button>
+        </td>
+        <td class="govuk-table__cell">
+          <button class="govuk-button govuk-!-margin-right-6 red-button" data-module="govuk-button" (click)="deleteUser(user.id)">
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.html:102]: think we can remove this click handler until we implement edit + delete

**Review by oliver-scott — COMMENTED** [review]
looks good, just some small suggestions. Think you might need to re-run the pipeline for the code coverage to update

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts (hunk 1 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:35]: removed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts (hunk 2 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:43]: removed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts (hunk 3 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:75]: removed

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts (hunk 4 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.ts:56]: done

File: AdminWebsite/AdminWebsite/Controllers/UserIdentityController.cs (hunk 5 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/Controllers/UserIdentityController.cs:57]: done

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.html (hunk 6 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.html:96]: done

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.html (hunk 7 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/manage-team/manage-team.component.html:102]: done

**Review by oliver-scott — APPROVED** [review]

**Pull request merged**